### PR TITLE
Dynamic timeout variables

### PR DESCRIPTION
**Pull request by harris21**

### CODE REVIEW

File: src/BowlerServiceProvider.php [hunk 1]
```diff
@@ -34,8 +35,11 @@ public function register()
         $rbmqPort = config('queue.connections.rabbitmq.port');
         $rbmqUsername = config('queue.connections.rabbitmq.username');
         $rbmqPassword = config('queue.connections.rabbitmq.password');
-        $this->app->bind(Connection::class, function () use ($rbmqHost, $rbmqPort, $rbmqUsername, $rbmqPassword) {
-            return new Connection($rbmqHost, $rbmqPort, $rbmqUsername, $rbmqPassword);
+        $rbmqConnectionTimeout = (int) config('queue.connections.rabbitmq.connection_timeout');
+        $rbmqReadWriteTimeout = (int) config('queue.connections.rabbitmq.read_write_timeout');
+        $rbmqHeartbeat = (int) config('queue.connections.rabbitmq.heartbeat');
```
Comment (Mulkave) [inline src/BowlerServiceProvider.php]: enforce the defaults from here, for if they're not set they'll be sent as empty and will override the default.

File: src/Connection.php [hunk 2]
```diff
@@ -81,11 +108,11 @@ public function __construct($host = 'localhost', $port = 5672, $username = 'gues
...
             $login_response = null,
             $locale = 'en_US',
-            $connection_timeout = 30,
-            $read_write_timeout = 30,
+            $connectionTimeout,
+            $readWriteTimeout,
             $context = null,
             $keepalive = false,
-            $heartbeat = 15
+            $heartbeat
```
Comment (Mulkave) [inline src/Connection.php]: missing configuration reading tests.

File: tests/ConnectionTest.php [hunk 3]
```diff
@@ -46,4 +46,16 @@ public function test_fetching_consumers_default()
...
+    public function test_set_configurations_default()
+    {
+        $heartbeat = $this->app['config']->get('queue.connections.rabbitmq.heartbeat');
+        $this->assertEquals(15, $heartbeat);
+
+        $connectionTimeout = $this->app['config']->get('queue.connections.rabbitmq.connection_timeout');
+        $this->assertEquals(30, $connectionTimeout);
+
+        $readWriteTimeout = $this->app['config']->get('queue.connections.rabbitmq.read_write_timeout');
+        $this->assertEquals(30, $readWriteTimeout);
```
Comment (Mulkave) [inline tests/ConnectionTest.php]: with this you're testing the functionality of the `Config` class, while what you should be testing is the service provider's initialisation of the `Connection` object, you can achieve this by mocking `AMQPStreamConnection`. You need to know which configuration params it used so there should be two tests:

1. connection created with default params
2. changing config params and verifying whether it creates the connection with the changed params

File: README.md [hunk 4]
```diff
@@ -34,17 +34,22 @@ Tools like the Rabbitmq *[Management](https://www.rabbitmq.com/management.html)*
...
-In order to configure rabbitmq host, port, username and password, add the following inside the connections array in config/queue.php file:
+In order to configure rabbitmq host, port, username, password and configure the timeout values too, add the following inside the connections array in config/queue.php file:
 
 ```php
 'rabbitmq' => [
     'host' => 'host',
     'port' => port,
     'username' => 'username',
     'password' => 'password',
+    'connection_timeout' => connection_timeout,
```
Comment (KinaneD) [inline README.md]: Please use integers here

**Pull request merged**